baremaps-core/src/main/java/org/apache/baremaps/vectortile/VectorTileEncoder.java

```diff
@@ -0,0 +1,360 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.baremaps.vectortile;
+
+import static org.apache.baremaps.vectortile.VectorTileUtils.*;
+
+import com.google.common.collect.Lists;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.function.Consumer;
+import org.apache.baremaps.mvt.binary.VectorTile;
+import org.locationtech.jts.geom.*;
+
+/**
+ * A vector tile encoder.
+ *
+ * This implementation is based on the Vector Tile Specification 2.1.
+ */
+public class VectorTileEncoder {
+
+  private int cx = 0;
+
+  private int cy = 0;
+
+  private List<String> keys = new ArrayList<>();
+
+  private List<Object> values = new ArrayList<>();
+
+  /**
+   * Constructs a vector tile encoder.
+   */
+  public VectorTileEncoder() {
+
+  }
+
+  /**
+   * Encodes a tile into a vector tile.
+   * 
+   * @param tile The tile to encode
+   * @return The vector tile
+   */
+  public VectorTile.Tile encodeTile(Tile tile) {
+    VectorTile.Tile.Builder builder = VectorTile.Tile.newBuilder();
+    tile.getLayers().forEach(layer -> builder.addLayers(encodeLayer(layer)));
+    return builder.build();
+  }
+
+  /**
+   * Encodes a layer into a vector tile layer.
+   * 
+   * @param layer The layer to encode
+   * @return The vector tile layer
+   */
+  public VectorTile.Tile.Layer encodeLayer(Layer layer) {
+    keys = new ArrayList<>();
+    values = new ArrayList<>();
+
+    VectorTile.Tile.Layer.Builder builder = VectorTile.Tile.Layer.newBuilder();
+    builder.setName(layer.getName());
+    builder.setVersion(2);
```

Comment:
Is this hard-coded value needed, or can it be a constant or embedded in the Builder ?

Reply:
Yes, we use the version 2 of the specification.